_includes/parse-server/class-level-permissions.md

```diff
@@ -28,4 +119,163 @@ If you want to restrict access to a full class to only authenticated users, you
 }
 ```
 
-Note that this is in no way securing your content. If you allow anyone to log in to your server, any client will be able to query this object.
+Effects:
+
+* Non authenticated users won't be able to do anything.
+* Authenticated users (any user with a valid sessionToken) will be able to read all the objects in that class
+* Users belonging to the admin role, will be able to perform all operations.
+
+⚠️ Note that this is in no way securing your content, if you allow anyone to log in to your server, any client will still be able to query this object.
+
+## Pointer Permissions
+
+While permissions discussed before let you explicitly target a user by id or a role,
+Pointer Permissions let you dynamically enforce permissions without knowing the id or assigning roles in advance. Instead you define one or more field names of this class, and any users pointed by such fields of a particular object are granted the permission.
+
+To use this feature, a field must:
+
+* already exist in this collection's schema
+* be of either `Pointer<_User>` or `Array` type
+
+In case of `Array`, only items that are of `Pointer<_User>` type are evaluated, other items silently ignored.
+
+You can think of it as a virtual ACL or a dynamic role defined per-object in its own field.
+
+There are two ways you can set Pointer Permission in schema:
+
+* [Using granular permissions](#granular-pointer-permissions) - `pointerFields` *requires Parse-Server v3.11 and above*
+* [Using grouped permissions](#grouped-pointer-permissions) - `readUserFields`/`writeUserFields`
+
+⚠️ `create` operation can't be allowed by pointer permissions, because there is literally no object to check it's field untill it is created;
+
+⚠️ `addField`  grants permission to only update an object with a new field, but it is advised to set addField permission using other means (e.g. restrict to a role or particular admin user by id).
```

Comment:
This brought me to a question - why do we have CLP section in Parse-Server guide in the first place?  It seems to have info about server as a binary, its setup and launch, while CLP is more about schema level feature. I suggest to remove CLP from server guide as a whole.

Reply:
good question, I think the Parse Server guide has been under loved so only contains limited info about basic / required setup - and there are discrepancies in what is included in the Parse Server README and the guide - I did aim to consolidate all the info into the guide but didn't get around to it.

I'm wondering whether we could remove the CLP section in the server guide in favour of adding the common security doc but I'm not sure how this would work in practice with the code sample etc?

Reply:
In fact the code samples are mostly schema configs - plain JSON objects. And a few examples come with several demo setup lines (that i've already translated to most sdk languages in this pr). 
Anyway clp and protected fields are used by database admin rather than client code or any of SDKs. I also think it could be very helpful and time saving if we maintain such examples as a shared postman collection.

Reply:
I see that.

I think adding the common security doc to the server guide would be good, that way there is only one 'copy' to be maintained and its available on all the guides which would encourage devs to think about security when they are implementing their client side.

Reply:
So are you happy with that or do you have other ideas?

I think it needs to be in there server guide in some way as it is related to setup - especially as you have gone to all the effort of writing it and I don't think it would make sense to have it in the client guides but not the server one?

Reply:
I agree we should merge it and only have a single copy as you proposed. As for Parse-Server Guide probably REST-flavored examples should fit best.

Reply:
Cool, REST example would make sense. I'll leave that with you then, or if you've got maintainer commits turned on I could take a look later.

Reply:
I believe i created this pr with edits enabled. Not sure if there is anything else i should allow?
<img width="261" alt="image" src="https://user-images.githubusercontent.com/2188841/75628552-3c941000-5beb-11ea-9d89-d98ef63d0181.png">
 I'm planning to work on it later today anyway

Reply:
Yes that’s the one, the annoying thing is that on GitHub there is no indication to me whether you have that ticked or not.